tests/schema/test_input.py

```diff
@@ -27,3 +29,72 @@ def filter(self, input: FilterInput) -> str:
     assert not result.errors
     assert result.data
     assert result.data["filter"] == "Hello nope"
+
+
+def test_input_with_nonscalar_field_default():
```

Comment:
I originally thought this had issues, but as long as you specify the default factory, this works as intended.  I didn't see any tests explicitly testing this so I added one.